### ISSUE TITLE
[Bug] Shell TitleView render issues

### ISSUE THREAD

**Issue by bcaceiro** [xamarin/Xamarin.Forms#13561]
### Description
After upgrading to the latest Xamarin.Forms version (5.0.0.1931), Shell TitleView is not correctly rendered when using Grids.

### Steps to Reproduce

1. Run Sample
2. Change between commented lines in ItemDetailPage ( show Label or Grid + Label)

### Expected Behavior
![Screenshot 2021-01-27 at 16 18 07](https://user-images.githubusercontent.com/2139254/106020444-ae0a6600-60bb-11eb-8e7f-6e6e77a9b93b.png)

### Actual Behavior
![Screenshot 2021-01-27 at 16 18 53](https://user-images.githubusercontent.com/2139254/106020466-b793ce00-60bb-11eb-9630-611d3d3d4e2a.png)

### Basic Information

- Version with issue:
- Last known good version: 4.8
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
- NuGet Packages: Xamarin.Essentials
- Affected Devices: iPhones with Notch

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Community 2019 for Mac ===

Version 8.8.5 (build 18)
Installation UUID: 46202649-1fdb-4dad-8746-d50e2e4c19c1
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000113

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.113 (2020-02/4fdfb5b1fd5) (64-bit)
	Package version: 612000113

=== Roslyn (Language Service) ===

3.8.0-5.20519.18+4c195c3ac1974edcefa76774d7a59a2350ec55fa

=== NuGet ===

Version: 5.8.0.6860

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/5.0.102/Sdks
SDK Versions:
	5.0.102
	5.0.101
	5.0.100
	3.1.405
	3.1.404
	3.1.403
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	5.0.2
	5.0.1
	5.0.0
	3.1.11
	3.1.10
	3.1.9

=== .NET Core 3.1 SDK ===

SDK: 3.1.405

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 11.1.0.26 (Visual Studio Community)
Commit: xamarin-android/d16-8/a36ce73
Android SDK: /Users/bcaceiro/Library
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.5
SDK Build Tools Version: 30.0.3

Build Information: 
Mono: 5e9cb6d
Java.Interop: xamarin/java.interop/d16-8@79d9533
ProGuard: Guardsquare/proguard/proguard6.2.2@ebe9000
SQLite: xamarin/sqlite/3.32.1@1a3276b
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-8@2fb1cbc

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/bcaceiro/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.8.0.32
Hash: 01a7774
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:24 UTC

=== Android Device Manager ===

Version: 16.8.0.46
Hash: 0a81419
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:32:44 UTC

=== Apple Developer Tools ===

Xcode 12.3 (17715)
Build 12C33

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.8.0.3 (Visual Studio Community)
Hash: c51fabee8
Branch: xcode12.3
Build date: 2020-12-10 21:05:05-0500

=== Xamarin Designer ===

Version: 16.8.0.510
Hash: 44e3f3ce9
Branch: remotes/origin/d16-8
Build date: 2020-12-10 00:06:14 UTC

=== Build Information ===

Release ID: 808050018
Git revision: e83d51ec050ffcdd573462fc1889b58cfe1a5ad5
Build date: 2021-01-07 14:45:50-05
Build branch: release-8.8
Xamarin extensions: e83d51ec050ffcdd573462fc1889b58cfe1a5ad5

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.1.0 Darwin Kernel Version 20.1.0
    Sat Oct 31 00:07:11 PDT 2020
    root:xnu-7195.50.7~2/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

XAML Styler 2.0.1


```

</details>

### Reproduction Link

https://we.tl/t-borESteDpy

**Comment by rachelkang** [comment]
Hi, @bcaceiro - thanks for submitting this issue! This issue will be fixed by #13514 :)

Note: was also able to repro on iOS devices that do not have the notch

**Comment by bcaceiro** [comment]
Any news on this issue? It was marked closed, but the problem remains with the latest Xamarin.Forms version..
